packages/uniforms-bridge-zod/__tests__/ZodBridge.ts

```diff
@@ -0,0 +1,665 @@
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getField('a')).toBe(schema.shape.a);
+      expect(bridge.getField('a.$')).toBe(schema.shape.a.element);
+      expect(bridge.getField('a.$.$')).toBe(schema.shape.a.element.element);
+    });
+
+    it('works with nested objects', () => {
+      const schema = object({ a: object({ b: object({ c: string() }) }) });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getField('a')).toBe(schema.shape.a);
+      expect(bridge.getField('a.b')).toBe(schema.shape.a.shape.b);
+      expect(bridge.getField('a.b.c')).toBe(schema.shape.a.shape.b.shape.c);
+    });
+
+    it('works with default', () => {
+      const schema = object({ a: object({ b: string() }).default({ b: 'x' }) });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getField('a')).toBe(schema.shape.a);
+      expect(bridge.getField('a.b')).toBe(
+        schema.shape.a.removeDefault().shape.b,
+      );
+    });
+
+    it('works with optional', () => {
+      const schema = object({ a: optional(object({ b: string() })) });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getField('a')).toBe(schema.shape.a);
+      expect(bridge.getField('a.b')).toBe(schema.shape.a.unwrap().shape.b);
+    });
+  });
+
+  describe('#getInitialValue', () => {
+    it('works with array', () => {
+      const schema = object({ a: array(array(string())) });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getInitialValue('a')).toEqual([]);
+      expect(bridge.getInitialValue('a.0')).toEqual([]);
+      expect(bridge.getInitialValue('a.0.0')).toEqual(undefined);
+    });
+
+    it('works with array (min length)', () => {
+      const schema = object({ a: array(array(string()).min(1)).min(2) });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getInitialValue('a')).toEqual([[], []]);
+      expect(bridge.getInitialValue('a.0')).toEqual([]);
+      expect(bridge.getInitialValue('a.0.0')).toEqual(undefined);
+    });
+
+    it('works with boolean', () => {
+      const schema = object({ a: boolean() });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getInitialValue('a')).toEqual(undefined);
+    });
+
+    it('works with date', () => {
+      const schema = object({ a: date() });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getInitialValue('a')).toEqual(undefined);
+    });
+
+    it('works with enum (array)', () => {
+      const schema = object({ a: enum_(['x', 'y', 'z']) });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getInitialValue('a')).toEqual('x');
+    });
+
+    it('works with enum (native, numbers)', () => {
+      enum Test {
+        x,
+        y,
+        z = 'a',
+      }
+
+      const schema = object({ a: nativeEnum(Test) });
+      const bridge = new ZodBridge(schema);
+      expect(bridge.getInitialValue('a')).toEqual('x');
+    });
```

Comment:
I just wanted to make sure it works with mixed values - strings are tested separately.

Reply:
Actually, you're right - it doesn't work at all.

Reply:
Due to the way the enums work, I had to remove the support for mixed enums completely - if there's at least one number, we treat it as an number-only enum and string-only one otherwise.